client/my-sites/google-my-business/index.js

```diff
@@ -14,23 +14,42 @@ import { navigation, sites, siteSelection } from 'my-sites/controller';
 import { newAccount, selectBusinessType, selectLocation, stats } from './controller';
 import { getSelectedSiteId } from 'state/ui/selectors';
 import { getKeyringConnectionsByName } from 'state/sharing/keyring/selectors';
-import {
-	getGoogleMyBusinessConnectedLocation,
-	getGoogleMyBusinessLocations,
-} from 'state/selectors';
+import { isGoogleMyBusinessLocationConnected, getGoogleMyBusinessLocations } from 'state/selectors';
 import { requestSiteSettings } from 'state/site-settings/actions';
 import { requestKeyringConnections } from 'state/sharing/keyring/actions';
 
+const redirectToStatsIfLocationConnected = ( context, next ) => {
+	const siteId = getSelectedSiteId( context.store.getState() );
+	context.store
+		.dispatch( requestSiteSettings( siteId ) )
+		.then( () => {
+			if ( isGoogleMyBusinessLocationConnected( context.store.getState(), siteId ) ) {
+				page.redirect( `/google-my-business/stats/${ context.params.site }` );
+			} else {
+				next();
+			}
+		} )
+		.catch( next ); // on error, try to move forward
+};
+
 export default function( router ) {
 	router( '/google-my-business', siteSelection, sites, navigation, makeLayout );
 
 	if ( config.isEnabled( 'google-my-business' ) ) {
-		router( '/google-my-business/new', redirectLoggedOut, siteSelection, sites, makeLayout );
+		router(
+			'/google-my-business/new',
+			redirectLoggedOut,
+			siteSelection,
+			redirectToStatsIfLocationConnected,
+			sites,
+			makeLayout
+		);
 
 		router(
 			'/google-my-business/new/:site',
 			redirectLoggedOut,
 			siteSelection,
+			redirectToStatsIfLocationConnected,
```

Comment:
I think we should still be able to access new even if we have a connected location (in case we want to create a new one). Granted it's not super important as the form to create a new one isn't there yet

Reply:
I'm pinging here @stephanethomas as he suggested it,
I'll remove it for now and we'll loop back to it if Stéphane thinks it's required.